### ISSUE TITLE
Services randomly get connect timeout

### ISSUE THREAD

**Issue by Kealman** [moby/moby#38365]
**Description**
Randomly get timeout between RabbitMQ and other services witch connected with it inside a swarm cluster.
We have a few services which connected to RabbitMQ (+ rabbitmq prometeus exporter), but half of them live outside the cluster, and another half inside. The inside services from time to time get timeout with connect to rabbitmq (include the exporter) but the outside service (through ingress network) at the same time don't have any connection issue and work well. It happens only with RabbitMQ, also there is ElasticSearch inside the cluster, but it don't have any issues like that, a services which use it work well all the time. 
I'm not sure that this is the swarm issue, but I'd be thankful for any help to investigate it.

**Steps to reproduce the issue:**
It's a little complicated provide the steps, because it happens randomly

**Describe the results you received:**
A services inside the cluster get timeout with connect to RabbitMQ

**Describe the results you expected:**
A services should keep connect

**Additional information you deem important (e.g. issue happens only occasionally):**
Log from rabbitmq exporter (https://github.com/kbudde/rabbitmq_exporter)
```
time="2018-12-13T07:47:12Z" level=info msg="Metrics updated" duration=43.796899ms,
time="2018-12-13T07:47:47Z" level=error msg="Error while retrieving data from rabbitHost" error="Get http://rabbitmq:15672/api/exchanges: net/http: request canceled (Client.Timeout exceeded while awaiting headers)" host="http://rabbitmq:15672" statusCode=0,
time="2018-12-13T07:47:47Z" level=info msg="Metrics updated" duration=30.025137648s,
time="2018-12-13T07:47:47Z" level=info msg="Metrics updated" duration=20.167315ms,
time="2018-12-13T07:47:07Z" level=info msg="Metrics updated" duration=38.563569ms
```

Sometimes look like that
```

time="2018-12-13T10:29:37Z" level=info msg="Metrics updated" duration=31.156976ms,
time="2018-12-13T10:29:36Z" level=info msg="Metrics updated" duration=4.40169636s,
time="2018-12-13T10:29:36Z" level=error msg="Error while retrieving data from rabbitHost" error="Get http://rabbitmq:15672/api/queues: dial tcp: lookup rabbitmq on 127.0.0.11:53: no such host" host="http://rabbitmq:15672" statusCode=0,
time="2018-12-13T10:29:36Z" level=error msg="Error while retrieving data from rabbitHost" error="Get http://rabbitmq:15672/api/overview: dial tcp: lookup rabbitmq on 127.0.0.11:53: no such host" host="http://rabbitmq:15672" statusCode=0,
time="2018-12-13T10:29:36Z" level=error msg="Error while retrieving data from rabbitHost" error="Get http://rabbitmq:15672/api/nodes: dial tcp: lookup rabbitmq on 127.0.0.11:53: no such host" host="http://rabbitmq:15672" statusCode=0,
time="2018-12-13T10:29:36Z" level=error msg="Error while retrieving data from rabbitHost" error="Get http://rabbitmq:15672/api/exchanges: dial tcp: lookup rabbitmq on 127.0.0.11:53: no such host" host="http://rabbitmq:15672" statusCode=0,
time="2018-12-13T10:29:27Z" level=info msg="Metrics updated" duration=44.632409ms
```

**Output of `docker version`:**
From node where live RabbitMQ
```
bash-4.3# docker version
Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        e68fc7a
 Built:             Tue Aug 21 17:20:42 2018
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       e68fc7a
  Built:            Tue Aug 21 17:29:02 2018
  OS/Arch:          linux/amd64
  Experimental:     true
```

**Output of `docker info`:**
From node where live RabbitMQ
```
bash-4.3# docker info
Containers: 12
 Running: 12
 Paused: 0
 Stopped: 0
Images: 13
Server Version: 18.06.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: active
 NodeID: hc70237firjwyh98t3fo9qg2z
 Is Manager: true
 ClusterID: xzdflj27az2e6vd3s4lxl0q4j
 Managers: 3
 Nodes: 5
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 10
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
  Force Rotate: 0
 Autolock Managers: false
 Root Rotation In Progress: false
 Node Address: 172.31.14.144
 Manager Addresses:
  172.31.14.144:2377
  172.31.22.19:2377
  172.31.41.23:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 468a545b9edcd5932818eb9de8e72413e616e86e
runc version: 69663f0bd4b60df09991c08812a60108003fa340
init version: fec3683
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.114-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.853GiB
Name: ip-172-31-14-144.eu-west-1.compute.internal
ID: W5DM:P3UD:TDGZ:YPZI:WMBE:CBHM:ZBN6:Z6WJ:UUVS:GXT7:KUWP:6QWU
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
 os=linux
 region=eu-west-1
 availability_zone=eu-west-1a
 instance_type=t2.medium
 node_type=manager
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
We're using Docker for AWS setup 
https://docs.docker.com/docker-for-aws/

**Comment by olljanat** [comment]
@Kealman which RabbitMQ version you are using? We have very similar setup but I have not seen issues like these.

**Comment by olljanat** [comment]
Derek add label: status/more-info-needed

**Comment by Kealman** [comment]
> @Kealman which RabbitMQ version you are using? We have very similar setup but I have not seen issues like these.
@olljanat 
We are using version: 3.7.8

**Comment by Kealman** [comment]
@olljanat Thank you for your answer! Sorry for long response. Actually, that problem was solved when we rolled back to `18.03 AWS for docker`. It was production environment so we could not be able to have enough experiments :(
But as soon as we can we are going to investigate this, and if we find smth I will post it here to help other who caught this :)
